### PR TITLE
array / it-academy.by / M-JC1-69-21-8 / Igor Popov / Done tasks 1-3

### PR DESCRIPTION
**Pull request by ipopov91**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/array/Task02.java [hunk 1]
```diff
@@ -31,9 +34,17 @@ public static void main(String[] args) {
...
      */
     static double getAverageMark(int[][] marks) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return 0.0;
+        int marksAverage = 0;
+        int marksSum = 0;
+
+        for (int i = 0; i < marks.length; i++) {
+            for (int j = 0; j < marks[i].length; j++) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/array/Task02.java:41]: Best way: using `foreach`

File: src/main/java/com/rakovets/course/java/core/practice/array/Task03.java [hunk 2]
```diff
@@ -33,9 +35,15 @@ public static void main(String[] args) {
      * @return средняя арифметическая отметка
      */
     static double[] getAverageMarks(int[][] marks) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+        double[] marksAverage = new double[marks.length];
+
+        for (int i = 0; i < marks.length; i++) {
+            for (int j = 0; j < marks[0].length; j++) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/array/Task03.java:41]: Best way: using `foreach`

**Review by rakovets — CHANGES_REQUESTED** [review]
You need to fix the following problems

**Review by rakovets — CHANGES_REQUESTED** [review]
You need to fix the following problems and add solution with completed tests

File: src/main/java/com/rakovets/course/java/core/practice/array/Task02.java (hunk 1 above)
Comment (ipopov91) [reply src/main/java/com/rakovets/course/java/core/practice/array/Task02.java]: done

File: src/main/java/com/rakovets/course/java/core/practice/array/Task03.java (hunk 2 above)
Comment (ipopov91) [reply src/main/java/com/rakovets/course/java/core/practice/array/Task03.java]: done

File: src/main/java/com/rakovets/course/java/core/practice/array/Task03.java [hunk 3]
```diff
@@ -45,9 +53,17 @@ public static void main(String[] args) {
...
      */
     static int[] getMinMarks(int[][] marks) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+        int[] minMark = new int [marks.length];
+
+        for (int i = 0; i < marks.length; i++) {
+            minMark[i] = marks[i][0];
+            for (int j = 0; j < marks[0].length; j++) {
```
[the remark this answers is not in the document]
Comment (ipopov91) [reply src/main/java/com/rakovets/course/java/core/practice/array/Task03.java]: done

File: src/main/java/com/rakovets/course/java/core/practice/array/Task03.java [hunk 4]
```diff
@@ -57,8 +73,16 @@ public static void main(String[] args) {
...
      */
     static int[] getMaxMarks(int[][] marks) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+        int[] maxMark = new int [marks.length];
+
+        for (int i = 0; i < marks.length; i++) {
+            maxMark[i] = marks[i][0];
+            for (int j = 0; j < marks[0].length; j++) {
```
[the remark this answers is not in the document]
Comment (ipopov91) [reply src/main/java/com/rakovets/course/java/core/practice/array/Task03.java]: done

**Review by rakovets — APPROVED** [review]
Approved

**Pull request closed without merging**